zulipterminal/ui_tools/boxes.py

```diff
@@ -205,18 +205,16 @@ def _setup_common_private_compose(self) -> None:
         ]
         self.focus_position = self.FOCUS_CONTAINER_MESSAGE
 
-    def private_box_view(
-        self,
-        *,
-        recipient_user_ids: Optional[List[int]] = None,
+    def convert_id_to_info(
+        self, recipient_user_ids: Optional[List[int]] = None
```

Comment:
Would this be better incorporated into `_set_regular_and_typing_recipient_user_ids`?

Reply:
It would be possible if we were only talking about `convert_id_to_info` but `_set_regular_and_typing_recipient_user_ids` is also used by `update_recipients` function .

Reply:
OK, so
- `_set_regular_and_typing_recipient_user_ids` does the following, dependent upon passed user_ids
  - sets `recipient_user_ids` (autocompletion)
  - sets `typing_recipient_user_ids` (not including user)
- `update_recipients` (which appears to effectively be `update_recipients_from_emails_in_widget` or similar):
  - sets `recipient_emails`
  - calls `_set_regular_and_typing_recipient_user_ids` with calculated ids from recipient_emails
- `convert_id_to_info` does the following, dependent upon the passed user_ids
  - calls `_set_regular_and_typing_recipient_user_ids`
  - sets `recipient_emails`
  - sets `recipient_info`

This pattern suggests that the last two methods have a similar intent and so should share a name prefix, and generally could benefit from renaming.

The latter sets an extra attribute, which seems to be for initialization only - can we make that a return value instead?

Reply:
OK, so
- `_set_regular_and_typing_recipient_user_ids` does the following, dependent upon passed user_ids
  - sets `recipient_user_ids` (autocompletion)
  - sets `typing_recipient_user_ids` (not including user)
- `update_recipients` (which appears to effectively be `update_recipients_from_emails_in_widget` or similar):
  - sets `recipient_emails`
  - calls `_set_regular_and_typing_recipient_user_ids` with calculated ids from recipient_emails
- `convert_id_to_info` does the following, dependent upon the passed user_ids
  - calls `_set_regular_and_typing_recipient_user_ids`
  - sets `recipient_emails`
  - sets `recipient_info`

This pattern suggests that the last two methods have a similar intent and so should share a name prefix, and generally could benefit from renaming.

The latter sets an extra attribute, which seems to be for initialization only - can we make that a return value instead?